dml_events.go

```diff
@@ -406,17 +410,21 @@ func Int64Value(value interface{}) (int64, bool) {
 //
 // ref: https://github.com/mysql/mysql-server/blob/mysql-5.7.5/mysys/charset.c#L963-L1038
 // ref: https://github.com/go-sql-driver/mysql/blob/9181e3a86a19bacd63e68d43ae8b7b36320d8092/utils.go#L717-L758
-func appendEscapedString(buffer []byte, value string) []byte {
+func appendEscapedString(buffer []byte, value string, padToLength uint) []byte {
 	buffer = append(buffer, '\'')
 
-	for i := 0; i < len(value); i++ {
+	var i int
+	for i = 0; i < len(value); i++ {
 		c := value[i]
 		if c == '\'' {
 			buffer = append(buffer, '\'', '\'')
 		} else {
 			buffer = append(buffer, c)
 		}
 	}
+	for ; i < int(padToLength); i++ {
+		buffer = append(buffer, 0)
```

Comment:
Question: Appending 0 is OK here? Don't need something like `\0`? I ask because this buffer is eventually going to be turned into a `string`.

Reply:
`buffer` is a `[]byte` . As such, 0 seems the right thing to me

Reply:
Yes, but `buffer` eventually gets turned into a `string`, see https://github.com/Shopify/ghostferry/blob/e3f52d4/dml_events.go#L286. I'm not certain if that's the right encoding for the query as I'd assume MySQL wants you to specify `\0` in a statement, especially since we're not using prepared statements here.

Reply:
sorry, I saw your comment too late and I just pushed the new patch-set. Let me check a bit more - I thought the binary conversion would end up being the same thing

Reply:
admittedly: I'm coming from a c/c++ and Python background, so I don't know if it really makes a difference

in golang, `0` is a `int` while `'\x00'` is a `int32` but they should be the same value when encoding. But I have no problem using `'\x00'` if that's cleaner